crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -1403,38 +1390,58 @@ mod tests {
         loop {
             threads.retain(|thread| !thread.is_finished());
             entities.flush_as_invalid();
-            entities.entities_hot_for_remote = rng.r#gen::<u32>() & 127;
+            if RAND {
+                entities.entities_hot_for_remote = rng.r#gen::<u32>() & 127;
+            }
             if threads.is_empty() {
                 break;
             }
             if timeout.elapsed().as_secs() > 60 {
-                panic!("remote entities timmed out.")
+                panic!("remote entities timed out.")
             }
         }
 
-        // It might be a little over since we may have reserved extra entities for remote reservation.
-        assert!(entities.len() >= 300);
         assert_eq!(
-            entities.remote.keep_hot.load(Ordering::Relaxed),
+            entities.remote.in_channel.load(Ordering::Relaxed),
             entities.remote.reserved.len() as u32
         );
     }
 
     #[test]
     #[cfg(feature = "std")]
-    fn remote_reservation() {
+    fn remote_reservation_empty_hot() {
+        let mut entities = Entities::new();
+        entities.entities_hot_for_remote = 0;
+        test_remote_reservation::<false>(&mut entities);
+    }
+
+    #[test]
+    #[cfg(feature = "std")]
+    fn remote_reservation_standard_hot() {
         let mut entities = Entities::new();
         // Lower batch size so more waiting is tested.
         entities.entities_hot_for_remote = 16;
-        test_remote_reservation(&mut entities);
+        test_remote_reservation::<false>(&mut entities);
+    }
+
+    #[test]
+    #[cfg(feature = "std")]
+    fn remote_reservation_volitol_hot() {
```

Comment:
I'm not sure, but `volitol` -> `volatile`? Maybe another word could be `frequently_changed`?